pkg/cmd/serve.go

```diff
@@ -57,8 +62,18 @@ func RegisterServeFlags(cmd *cobra.Command, config *server.Config) {
 
 	// Flags for configuring the dispatch server
 	util.RegisterGRPCServerFlags(cmd.Flags(), &config.DispatchServer, "dispatch-cluster", "dispatch", ":50053", false)
-	server.RegisterCacheConfigFlags(cmd.Flags(), &config.DispatchCacheConfig, "dispatch-cache")
-	server.RegisterCacheConfigFlags(cmd.Flags(), &config.ClusterDispatchCacheConfig, "dispatch-cluster-cache")
+	server.RegisterCacheFlags(cmd.Flags(), "dispatch-cache", &config.DispatchCacheConfig, &server.CacheConfig{
+		Enabled:     true,
+		Metrics:     false,
+		NumCounters: 10_000,
+		MaxCost:     "30%",
+	})
+	server.RegisterCacheFlags(cmd.Flags(), "dispatch-cluster-cache", &config.ClusterDispatchCacheConfig, &server.CacheConfig{
+		Enabled:     true,
+		Metrics:     false,
+		NumCounters: 100_000,
+		MaxCost:     "70%",
+	})
```

Comment:
I remember discussing we could have this cache be shared between "dispatch cluster cache" and "cluster cache". Is that something that will be addressed in a follow up?

Reply:
We're putting that out of scope for now. I've downscoped this to the original goal which is to let folks specify percentages of available memory.